xtask/src/ci/target_matrix.rs

```diff
@@ -1,20 +1,30 @@
+use std::process::Command;
+
 use clap::builder::BoolishValueParser;
 use clap::Parser;
-use serde::Serialize;
+use cross::{shell::Verbosity, CommandExt};
+use serde::{Deserialize, Serialize};
 
 use crate::util::{get_matrix, gha_output, gha_print, CiTarget, ImageTarget};
 
 pub(crate) fn run(message: String, author: String) -> Result<(), color_eyre::Report> {
     let mut matrix: Vec<CiTarget> = get_matrix().clone();
-    if author == "bors[bot]" && message.starts_with("Try #") {
-        if let Some((_, args)) = message.split_once(": ") {
-            let app = TargetMatrixArgs::parse_from(args.split(' '));
-            app.filter(&mut matrix);
-        }
+    let (prs, mut app) = if author == "bors[bot]" {
+        process_bors_message(&message)?
     } else {
-        gha_print("Running all targets.");
+        (vec![], TargetMatrixArgs::default())
+    };
+
+    if !prs.is_empty()
+        && prs.iter().try_fold(true, |b, pr| {
+            Ok::<_, eyre::Report>(b && has_no_ci_target(*pr)?)
+        })?
+    {
+        app.none = true;
     }
 
+    app.filter(&mut matrix);
```

Comment:
this could be consolidated, maybe?

Reply:
I think it's fine. The logic is clear, and that's what matters.